Language service reports error on "export import" use

Using the TypeScript 1.6 beta in an AMD project with the following structure:

In one.ts:
export var something = 7;

In two.ts:
export import Something = require("one");

The project compiles successfully, but the language service for Visual Studio 2013 reports an error "Cannot find module 'one'." in two.ts.

--- a/src/services/services.ts
+++ b/src/services/services.ts
@@ -2152,6 +2152,7 @@ namespace ts {
             //
             //    export * from "mod"
             //    export {a as b} from "mod"
+            //    export import i = require("mod")
 
             while (token !== SyntaxKind.EndOfFileToken) {
                 if (token === SyntaxKind.DeclareKeyword) {
@@ -2273,6 +2274,25 @@ namespace ts {
                             if (token === SyntaxKind.StringLiteral) {
                                 // export * from "mod"
                                 recordModuleName();
+                            }
+                        }
+                    }
+                    else if (token === SyntaxKind.ImportKeyword) {
+                        token = scanner.scan();
+                        if (token === SyntaxKind.Identifier || isKeyword(token)) {
+                            token = scanner.scan();
+                            if (token === SyntaxKind.EqualsToken) {
+                                token = scanner.scan();
+                                if (token === SyntaxKind.RequireKeyword) {
+                                    token = scanner.scan();
+                                    if (token === SyntaxKind.OpenParenToken) {
+                                        token = scanner.scan();
+                                        if (token === SyntaxKind.StringLiteral) {
+                                            //  export import i = require("mod");
+                                            recordModuleName();
+                                        }
+                                    }
+                                }
                             }
                         }
                     }

--- a/tests/cases/unittests/services/preProcessFile.ts
+++ b/tests/cases/unittests/services/preProcessFile.ts
@@ -173,6 +173,20 @@ describe('PreProcessFile:', function () {
                    isLibFile: false
                })  
         });
+
+        it("Correctly handeles export import declarations", function () {
+            test("export import a = require(\"m1\");",
+                true,
+                {
+                    referencedFiles: [],
+                    importedFiles: [
+                        { fileName: "m1", pos: 26, end: 28 }
+                    ],
+                    ambientExternalModules: undefined,
+                    isLibFile: false
+                })
+        });
+
     });
 });
 

